x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -130,10 +131,14 @@ protected void computeScores(RankFeatureDoc[] featureDocs, ActionListener<float[
      */
     @Override
     protected RankFeatureDoc[] preprocess(RankFeatureDoc[] originalDocs) {
-        return Arrays.stream(originalDocs)
-            .filter(doc -> minScore == null || doc.score >= minScore)
-            .sorted(Comparator.comparing((RankFeatureDoc doc) -> doc.score).reversed())
-            .toArray(RankFeatureDoc[]::new);
+        List<RankFeatureDoc> docs = new ArrayList<>();
+        for (RankFeatureDoc doc : originalDocs) {
+            if (minScore == null || doc.score >= minScore) {
+                doc.score = normalizeScore(doc.score);
+                docs.add(doc);
+            }
+        }
+        return docs.stream().sorted(Comparator.comparing((RankFeatureDoc doc) -> doc.score).reversed()).toArray(RankFeatureDoc[]::new);
```

Comment:
```
docs.sort(RankFeatureDoc::compareTo);
return docs.toArray(new RankFeatureDoc[0]);
```
Why not this? Seems way easier to read and avoids streams, which we know can cause way too much garbage.